src/Stratis.Bitcoin.Features.MemoryPool/MempoolOrphans.cs

```diff
@@ -151,17 +151,20 @@ public class OrphanTx
                     // might be now valid, e.g. due to a nLockTime'd tx becoming valid,
                     // or a double-spend. Reset the rejects filter and give those
                     // txs a second chance.
+                    this.logger.LogTrace("Executing task to clear rejected transactions.");
                     this.hashRecentRejectsChainTip = this.chain.Tip.HashBlock;
                     this.recentRejects.Clear();
                 });
             }
 
             // Use pcoinsTip->HaveCoinsInCache as a quick approximation to exclude
             // requesting or processing some txs which have already been included in a block
-            return await this.MempoolLock.ReadAsync(() =>
-                                this.recentRejects.ContainsKey(trxid) ||
-                                this.memPool.Exists(trxid) ||
-                                this.mapOrphanTransactions.ContainsKey(trxid));
+            bool isTxPresent =  this.recentRejects.ContainsKey(trxid) ||
```

Comment:
Just to confirm - add a local lock for the recenRejects,mapOrphanTransactions collections?

Reply:
Ok - I've done that. We may want to review the locking in general in mempool orphans as there may be other instances that can use the local lock instead of the mempool manager one. Next PR perhaps?